### ISSUE TITLE
Smart notification about stocking

### ISSUE THREAD

**Issue by arijit1035** [IITIDIDX597/sp_2021_team2#25]
Goal:
Cabi is a modular kitchen storage system that helps people keep it tidy, efficient, flexible, and connected which seamlessly identifies needs, displays relevant details for cooking and eases the kitchen experience.

Hill:
Managers of shared spaces, automate the monitoring of kitchen inventory items in real time and adds out-of-stock items to an ordering system, which replaces all inventory items from the fastest delivery available so the guest can focus on meal preparation.

Subhill:
Users can set a recipe they want to cook and Cabi can show where in the storage the items are located. If an item is missing it will notify the managers or automatically order for the items to be delivered.

Story:
As a Cabi user
I want to be notified when I am low on an ingredient or another item needs to be replaced
So that I can I buy more

Story v4.0
As a person on a limited cooking schedule
I want to understand if there are any low stock of items that I require prior to cooking
So that I can procure the items myself if the delivery time is too long.

**Comment by akallish** [comment]
> So that I can I buy more

I believe you mean so Cabi can procure these ingredients so I do not have to?

**Comment by arijit1035** [comment]
@k8forz @SanhyongPark @walkoutthebox  
Edits I am making based on the recommendation, feel free to modify.
Story:
As a Cabi user
I want to be notified when I am low on an ingredient or another item needs to be replaced
So that I can I get the kitchen items procured to me before starting the cooking process where those items are required.

**Comment by walkoutthebox** [comment]
@arijit1035  @k8forz @SanhyongPark I remember Kait mentioning that the point of this was to have a way for users to procure and provide ingredients themselves. The reason for their providing the item could be because the replacement wouldn't be on time for their meal preparation, or they could just want to bring something to share with others. I updated the language based on that general thought. 

As a Cabi user
I want to understand whether or not low stock items that I require will be replenished by the time I need them
So that I can procure the items myself if necessary.

**Comment by arijit1035** [comment]
@SanhyongPark @k8forz @walkoutthebox I feel the words need to be corrected a little.
v3.0 As a Cabi user
I want to understand if there are any low stock of items that I require immediately for cooking
So that I can procure the items myself if the delivery time is too long.

**Comment by arijit1035** [comment]
@SanhyongPark @k8forz @walkoutthebox 
v4.0 
As a person on a limited cooking schedule
I want to understand if there are any low stock of items that I require prior to cooking
So that I can procure the items myself if the delivery time is too long.

**Comment by k8forz** [comment]
**V1 of Acceptance Criteria #1** 

**Given**
that I want to check the stock of certain ingredients I require 
**When**
I'm preparing my meal 
**Then**
I can order the missing ingredients through the Cabi interface to be delivered in a timely manner

**Comment by k8forz** [comment]
> **V1 of Acceptance Criteria #1**
> 
> **Given**
> that I want to check the stock of certain ingredients I require
> **When**
> I'm preparing my meal
> **Then**
> I can order the missing ingredients through the Cabi interface to be delivered in a timely manner

This kind of feels like I re-worded the story but what do y'all think @SanhyongPark @arijit1035 @walkoutthebox

**Comment by walkoutthebox** [comment]
@k8forz This feels like an activity that should occur prior to the user being in the Cabi space. I know we say "I require immediately for cooking" but I would imagine there would need to be some advanced notice in order for the user to procure the item themselves prior to cooking. 

Thoughts? @SanhyongPark @arijit1035 @k8forz : We may want to change the story to read, "I want to understand if there are any low stock of items that I require **_prior_** to cooking

**V2 of Acceptance Criteria 1**

**Scenario:**

As a person on a limited cooking schedule
I want to understand if there are any low stock of items that I require immediately for cooking
So that I can procure the items myself if the delivery time is too long.

**Given**

I'm a signed in Cabi app user

**When** 

I select the recipe I plan to prepare

**Then**

Cabi system shows me the list of required recipe items along with an inventory status, which allows me to understand whether or not the ingredients are or will be available for my meal preparation.

**Comment by akallish** [comment]
@walkoutthebox @k8forz This is great conversation. @walkoutthebox is identifying improvements to the story focus which is great and not unlike when the team had to define what each word meant in a hill statement. 

If there are too many details to a story and scenarios, then create more stories with separate acceptance criteria. 

Remember that when things don't trace back to the sub-hill or hill, they can be modified if the information overrides what the higher levels are saying or not saying.

**Comment by arijit1035** [comment]
@walkoutthebox I agree Julian, the story in the current form misses out on this detail. @k8forz @SanhyongPark  post your opinion here so that we have a common consensus.

**Comment by k8forz** [comment]
**V3 of Acceptance Criteria 1**

**Given**
I'm a signed in Cabi app user

**When**
I select the recipe I plan to prepare

**Then**
the Cabi system shows me the list of required recipe items along with an inventory status, which allows me to understand whether or not the ingredients are or will be available _in time_ for my meal preparation.

**Comment by arijit1035** [comment]
V3 of Acceptance Criteria 1

Given
I'm a person on a limited cooking schedule

When
I select the recipe I plan to prepare on Cabi App in advance

Then
the Cabi system shows me the list of required recipe items along with an inventory status, which allows me to understand whether or not the ingredients are or will be available in time for my meal preparation to help me make an informed decision.

**Comment by k8forz** [comment]
**V3 of Acceptance Criteria 1**

**Given**
I'm a person on a limited cooking schedule

**When**
I select the recipe I plan to prepare on Cabi App#53 in advance

**Then**
the Cabi system shows me the list of required recipe items along with an inventory status, which allows me to understand whether or not the ingredients are or will be available in time for my meal preparation to help me make an informed decision.

**Comment by arijit1035** [comment]
@k8forz @SanhyongPark @walkoutthebox this looks good and captures the story.
